### ISSUE TITLE
Error with JsonToFormDecoder

### ISSUE THREAD

**Issue by Ovascos** [FriendsOfSymfony/FOSRestBundle#2144]
I'm using JsonToFormDecoder with the following (minimal) config:
```jaml
fos_rest:
    body_listener:
        decoders:
            json: fos_rest.decoder.jsontoform
```
I'm running on FOS Rest bundle version 2.7.2, Symfony 4.4. on PHP 7.4.4

My toy controller:
```php
    /**
     * @Rest\Post("/foo")
     */
    public function postFooAction(Request $request)
    {
        return $this->json(['message' => 'foo']);
    }
```

When creating a POST request with content
```json
"foo"
```
I get a HTTP 500 error with message `Warning: Invalid argument supplied for foreach()` at position `friendsofsymfony/rest-bundle/Decoder/JsonToFormDecoder.php` at line `28`. Note that the request is valid Json. Therefore, built-in `json_decode` returns a string which cannot be iterated in the foreach loop at line 28.

I would rather expect a 4xx error, as JsonToFormDecoder does not know how to convert this.

It seems that changing line 51 of JsonToFormDecoder.php to `is_array($decodedData)` fixes the issue, however, as I'm rather new with Symfony, I'm not sure whether are any side-effects.

**Comment by xabbuh** [comment]
Good catch @Ovascos. #2145 should fix this.

**Comment by xabbuh** [comment]
fixed in 2.7.3
